libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -678,6 +678,32 @@ string YulUtilFunctions::arrayDataAreaFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::storageArrayIndexAccessFunction(ArrayType const& _type)
+{
+	string functionName = "storage_array_index_access_" + _type.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(data, index) -> slot, offset {
+				<?multipleItemsPerSlot>
+				// TODO
+				<!multipleItemsPerSlot>
+					<?storageSizeNotOne>
```

Comment:
I think the optimizer can do that for us.

Reply:
Could you elaborate a bit please?

Reply:
Just keep `mul(x, <size>)`. If `size` is one, the optimizer will remove the multiplication.